File: 백준/Silver/25206. 너의 평점은/너의 평점은.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {

    public static float getScore(String grade) {
        if (grade.charAt(0) != 'F') {
            if (grade.charAt(1) == '+') {
                return 'F' - grade.charAt(0) - 0.5f;
            } else if (grade.charAt(1) == '0') {
                return 'F' - grade.charAt(0) - 1;
            }
        }
        return 0;
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        String grade;
        float hakjum, hakjumSum = 0;
        double allSum = 0;

        for (int i = 0; i < 20; i++) {
            String[] input = br.readLine().split(" ");
            hakjum = Float.parseFloat(input[1]);
            grade = input[2];
            if (grade.charAt(0) != 'P') {
                hakjumSum += hakjum;
                allSum += hakjum * getScore(grade);
            }
        }
        System.out.printf("%.6f\n", allSum / hakjumSum);

        br.close();
    }
}

File: 백준/Silver/25206. 너의 평점은/너의 평점은.py
import sys


def input(): return sys.stdin.readline().rstrip()


grades = {'A+': 4.5, 'A0': 4.0, 'B+': 3.5, 'B0': 3.0,
          'C+': 2.5, 'C0': 2.0, 'D+': 1.5, 'D0': 1.0, 'F': 0.0}

result = []
scores = []
for _ in range(20):
    _, score, grade = input().split()

    if grade == 'P':
        continue
    scores.append(float(score))

    result.append(float(score) * grades[grade])

print(sum(result) / sum(scores))